src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectRequest = "SELECT c.id AS car_id,model,manufacturer_id,m.name,country "
+                + "FROM CARS c "
+                + "JOIN MANUFACTURERS m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(selectRequest)) {
+
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+
+            if (resultSet.next()) {
+                car = parseCarWithManufacturerResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't get a Car by id = " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT *"
+                + "FROM CARS c "
+                + "JOIN MANUFACTURERS m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(getAllRequest)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get all cars from DB", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? and is_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateFieldsStatement =
+                        connection.prepareStatement(updateQuery);) {
+            updateFieldsStatement.setString(1, car.getModel());
+            updateFieldsStatement.setLong(2, car.getManufacturer().getId());
+            updateFieldsStatement.setLong(3, car.getId());
+            updateFieldsStatement.executeUpdate();
+
+            delete(car.getId());
+            insertDrivers(car);
```

Comment:
```suggestion
            delete(car.getId());
            insertDrivers(car);
```

after try/catch block